openfisca_core/tracers.py

```diff
@@ -224,3 +247,33 @@ def print_computation_log(self, aggregate = False):
         """
         for node, depth in self._computation_log:
             self._print_node(node, depth, aggregate)
+
+
+class TracingParameterNodeAtInstant():
+
+    def __init__(self, parameter_node_at_instant, tracer):
+        self.parameter_node_at_instant = parameter_node_at_instant
+        self.tracer = tracer
+
+    def __getattr__(self, key):
+        try:
+            child = self.parameter_node_at_instant._original_children[key]
+            period = self.parameter_node_at_instant._instant_str
+
+            if isinstance(child, ParameterNode):
+                return TracingParameterNodeAtInstant(child.get_at_instant(period), self.tracer)
+            else:
+                at_instant_trace = child.get_at_instant(period)
+                self.tracer.record_calculation_parameter_access(child, period)
+                return at_instant_trace
+        except KeyError:
+            param_name = self.parameter_node_at_instant._compose_name(self.parameter_node_at_instant._name, key)
+            raise ParameterNotFound(param_name, self._instant_str)
+    
+    def __getitem__(self, key):
+        return self.parameter_node_at_instant[key]
+        # If fancy indexing is used, cast to a vectorial node
```

Comment:
Tracing parameters call using fancy indexing seems non-trivial.
If fancy indexing is using on a leaf of the parameter tree, like `P.single.owner[zone]`, then your parameter value is just a vector, and you can use your regular call to `record_calculation_parameter_access`.
If it's used on nodes, like `P.single[housing_occupancy_status].z1`, that's harder, as you have a more complex object to insert a tracer into...

Reply:
Yes. We came to same conclusion with @Morendil: seems non-trivial. 😕 
Thanks a lot @guillett for recommending the test! 

So, as adding fancy indexed parameters in `/trace` will take some more time, we start by delivering this PR without them.

Reply:
Seems reasonable 🙂

Reply:
Is there a _real_ need to deliver that PR without _fancy indexing_? What is your definition of done from an end-user (who doesn't know when and where fancy-indexing is used)? For me, it makes sense to give another sprint' time.

Reply:
Trying to find a solution

Reply:
> Is there a real need to deliver that PR without fancy indexing?

YES!, 50 years of computer science and 20 years of agile software development have taught us that _divide and conquer_ is a good practice, and _scope creep_ is a bad one.

What I understand by _we **start** by delivering this PR without them_ is not that the issue is finished from an end-user perspective, only that letting scope creep and the pull request to rot increases the risk of this not being ever delivered.

> What is your definition of done from an end-user (who doesn't know when and where fancy-indexing is used)?

There is no such definition of done.

The proposal is to deliverer this incrementally and iteratively, which is a universal good development practice, not to abandon the effort half-way through, if that's what you're fearing.

> For me, it makes sense to give another sprint' time.

Sure. Which in no way prevent us from deliver.

Let's ship!

Reply:
Found a solution, we can actually use the same relatively simple mechanism for both cases.

Reply:
> Is there a real need to deliver that PR without fancy indexing?

Our reasoning was that fancy indexing is used in an extremely small fraction of all formulas, therefore the partial implementation delivers most of the value of the full one.